pkg/render/node.go

```diff
@@ -1086,7 +1086,7 @@ func (c *nodeComponent) nodeEnvVars() []v1.EnvVar {
 		// Add in Calico Enterprise specific configuration.
 		extraNodeEnv := []v1.EnvVar{
 			{Name: "FELIX_PROMETHEUSREPORTERENABLED", Value: "true"},
-			{Name: "FELIX_PROMETHEUSREPORTERPORT", Value: fmt.Sprintf("%d", nodeReporterPort)},
+			{Name: "FELIX_PROMETHEUSREPORTERPORT", Value: fmt.Sprintf("%d", c.nodeReporterMetricsPort)},
```

Comment:
Should we check that the port is not zero? I think it would probably be a problem if it was. Should we set the above line to false if it is zero?

Reply:
Or maybe we should check that it is not zero in the controller and degrade if it is? :shrug: